From 8b5eac78ce85fd326c59568d38ef4013b55e1b3e Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <nix.or.die@gmail.com>
Date: Mon, 15 Apr 2019 19:02:20 +0200
Subject: [PATCH 2/4] custom layout check for minSize > maxSize

 well this can be set , even if is by accident or miss calculation
 so check for it
---
 src/modules/partition/core/PartitionLayout.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/modules/partition/core/PartitionLayout.cpp b/src/modules/partition/core/PartitionLayout.cpp
index 963a6ceb1..9b4ac2eb9 100644
--- a/src/modules/partition/core/PartitionLayout.cpp
+++ b/src/modules/partition/core/PartitionLayout.cpp
@@ -135,6 +135,8 @@ PartitionLayout::execute( Device *dev, qint64 firstSector,
             size = minSize;
         if ( size > maxSize )
             size = maxSize;
+        if ( minSize > maxSize )
+            maxSize = minSize;
         if ( size > availableSize )
             size = availableSize;
         end = firstSector + size - 1;
-- 
2.21.0

